pubspec.yaml

```diff
@@ -1,21 +1,21 @@
 name: azure_application_insights
 description: A Dart client to integrate with Azure's Application Insights service
-version: 3.1.0
+version: 3.1.1
 repository: https://github.com/kentcb/azure_application_insights
 
 environment:
-  sdk: '>=2.12.0 <3.0.0'
+  sdk: '>=3.0.0 <4.0.0'
 
 dependencies:
   crypto: ^3.0.0
-  http: ^0.13.0
-  logging: ^1.0.2
-  meta: ^1.3.0
-  stack_trace: ^1.10.0
+  http: '>=0.13.6 <2.0.0'
```

Comment:
Why the change in syntax here? Can we just do `^1.0.0`?

Reply:
For sure you can go with a single entry `^1.0.0`. But then everybody who are using this package are forced to also use the http package version 1.0.0 - <2.0.0.
If you use `'>=0.13.6 <2.0.0'` , you will be compatible with http version 0.13.6 up to (but not including) 2.0.0.
In other words, but developers who must be on legacy 0.x.x can do so , as well as devs who are using the later 1.x.x

Specifying a single version (with caret):   `^0.13.6` is the same as `'>=0.13.6 <1.0.0'` (i.e. excluding v1.0.0).

Reference:
[Caret syntax](https://dart.dev/tools/pub/dependencies#caret-syntax)
[Versioning](https://dart.dev/tools/pub/versioning)

Reply:
Ah, I get it now. I thought that using SDK 3 would require the use of http 1.0.0 onwards, but I see that it's still possible to use older versions with newer SDKs.